modules/iot-core/main.tf

```diff
@@ -0,0 +1,81 @@
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#---------------------------------------------------------
+# Create IoT Core Registry
+#---------------------------------------------------------
+
+resource "google_cloudiot_registry" "registry" {
+
+  name    = var.registry_name
+  project = var.project_id
+  region  = var.region
+
+  dynamic "event_notification_configs" {
```

Comment:
if you use the suggested structure, this will look something like this
```
dynamic "event_notification_configs" {
   for_each = var.extra_telemetry_pub_sub_topic_ids
    content {
      pubsub_topic_name = each.key
      subfolder_matches = each.value
    }
  }
```